packages/commerce-sdk-react/src/hooks/useAsync.ts

```diff
@@ -24,21 +25,27 @@ export const useAsync = <T>(fn: () => Promise<T>, deps?: unknown[]): QueryRespon
     return result
 }
 
-export const useAsyncExecute = <A, R>(fn: (arg: A) => Promise<R>) => {
+export const useAsyncExecute = <A extends any[], R>(fn: (...args: A) => Promise<R>) => {
     // This is a stub implementation to validate the types.
     // The real implementation will be more React-y.
+    const [isLoading, setIsLoading] = useState<boolean>(false)
+    const [data, setData] = useState<R | undefined>(undefined)
+    const [error, setError] = useState<Error | undefined>(undefined)
     const result: ActionResponse<A, R> = {
-        isLoading: false,
-        execute(arg: A) {
-            result.isLoading = true
-            fn(arg)
-                .then((data) => {
-                    result.isLoading = false
-                    result.data = data
+        data,
+        error,
+        isLoading,
+        execute: (...arg) => {
+            setIsLoading(true)
+            fn(...arg)
+                .then((data: R) => {
```

Comment:
```suggestion
                .then((data) => {
```
Oh yeah, this type can be omitted because it's implicit!